docs/site/Binding.md

```diff
@@ -379,14 +382,154 @@ To understand the difference between `@bind()/@injectable` and `ctx.bind()`, see
 [Configure binding attributes for a class](#configure-binding-attributes-for-a-class).
 " %}
 
-### Refresh a binding with SINGLETON or CONTEXT scope
+There are some limitation of `SINGLETON`, `CONTEXT`, and `TRANSIENT` scopes.
+Consider the following typical context hierarchy formed by a LoopBack REST
+application:
 
-`SINGLETON` and `CONTEXT` scopes can be used to minimize the number of value
-instances created for a given binding. But sometimes we would like to force
-reloading of a binding when its configuration or dependencies are changed. For
-example, a logging provider can be refreshed to pick up a new logging level. The
-same functionality can be achieved with `TRANSIENT` scope but with much more
-overhead.
+```
+invocationCtx -> requestCtx -> serverCtx -> appCtx
+```
+
+We use `TRANSIENT` scope for controllers/services so that each request will get
+a new instance. But if a controller/service is resolved within the
+`invocationCtx` (by interceptors), a new instance will be created again.
```

Comment:
> a new instance will be created again.

Is it because each interceptor will resolve a controller/service instance?

Reply:
I improved the docs with an example to make it even more clear.